engine/crates/runtime/src/rate_limiting.rs

```diff
@@ -48,13 +49,27 @@ impl std::ops::Deref for RateLimiter {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
-pub struct RateLimitConfig {
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SubgraphRateLimitConfig {
     pub limit: usize,
     pub duration: Duration,
 }
 
 #[derive(Debug, Clone, Default)]
 pub struct KeyedRateLimitConfig<'a> {
-    pub rate_limiting_configs: HashMap<&'a str, RateLimitConfig>,
+    pub rate_limiting_configs: HashMap<&'a str, SubgraphRateLimitConfig>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct RateLimitRedisConfig<'a> {
```

Comment:
That shouldn't be part of `runtime` but in `runtime-local`, that's an implementation detail. The engine shouldn't be aware of this. Same for underneath